### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by ericpre**
### Progress of the PR
- [x] Fix deprecation scipy, matplotlib and numpy deprecation warnings,
- [x] Bump scipy from 1.5 to  1.6 to be able to use `scipy.integrate.simpson` instead of deprecated `scipy.integrate.simps`,
- [x] Fix warning in test suite,
- [x] Set `OMP_NUM_THREADS` as a default to run the test suite to avoid sklearn warning with MKL and possibly failure on Azure Pipelines CI (windows),
- [x] Tidy up some tests which were redundant,  
- [n/a] update docstring (if appropriate),
- [n/a] update user guide (if appropriate),
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [x] update tests,
- [x] ready for review.

### CODE REVIEW

**Comment by ericpre** [comment]
@jlaehne, @CSSFrancis, once this PR is merged, I would suggest to make a quick 2.0.1 release?

File: hyperspy/axes.py [hunk 1]
```diff
@@ -1264,11 +1264,13 @@ def value2index(self, value, rounding=round):
...
 
         if isinstance(value, np.ndarray):
+            if np.isnan(value).any():
+                raise ValueError("The value can't be 'Not a Number'.")
             index = index.astype(int)
             if np.all(self.size > index) and np.all(index >= 0):
                 return index
             else:
-                raise ValueError("A value is out of the axis limits")
+                raise ValueError("One of the value is out of the axis limits")
```
Comment (jlaehne) [inline hyperspy/axes.py]:
```suggestion
                raise ValueError("One of the values is out of the axis limits")
```

File: hyperspy/utils/plot.py [hunk 2]
```diff
@@ -29,7 +29,7 @@
 plot_histograms
     Compute and plot the histograms of multiple signals in the same figure.
 plot_roi_map
-    Plot a the spatial variation of a ROI within a `Signal`\ s signal axes.
+    Plot a the spatial variation of a signals sliced by a ROI in the signal space.
```
Comment (jlaehne) [inline hyperspy/utils/plot.py]:
```suggestion
    Plot a the spatial variation of a signal sliced by a ROI in the signal space.
```

**Review by jlaehne — APPROVED** [review]

**Comment by jlaehne** [comment]
One of the `Package & Test` workflows fails with `AttributeError: module 'mpmath' has no attribute 'rational'`

Seems to originate from a recent pre-release of `mpmath`: https://github.com/NVIDIA/TensorRT-LLM/issues/1145

**Comment by jlaehne** [comment]
> @jlaehne, @CSSFrancis, once this PR is merged, I would suggest to make a quick 2.0.1 release?

Fixed a few typos, otherwise very good idea to reduce the number of warnings.

**Comment by ericpre** [comment]
> One of the `Package & Test` workflows fails with `AttributeError: module 'mpmath' has no attribute 'rational'`
> 
> Seems to originate from a recent pre-release of `mpmath`: [NVIDIA/TensorRT-LLM#1145](https://github.com/NVIDIA/TensorRT-LLM/issues/1145)

Changed to install `mpmath` separately (without the `--pre` flag) as a workaround in https://github.com/hyperspy/.github/commit/b4c0b0951aefed778606a4276f5223f2d9713d48.

**Pull request merged**